migrations/Version20171122181453.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Migrations;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Changes in exercise and pipeline configuration which reflects changes in
+ * judge boxes interface.
+ */
+class Version20171122181453 extends AbstractMigration
+{
+  const JUDGE_TYPE = "judge";
+  const JUDGE_ARGS_PORT = "args";
+  const JUDGE_ARGS_VAR = "judge-args";
+  const JUDGE_ARGS_REF = "\$judge-args";
+  const CUSTOM_JUDGE_VAR = "custom-judge";
+  const JUDGE_TYPE_VAR = "judge-type";
+
+  /**
+   * Add newly defined ports in judge box to corresponding structures in
+   * database.
+   * @param $pipelineConfig
+   */
+  private function processPipeline($pipelineConfig) {
+    // prep
+    $config = Yaml::parse($pipelineConfig["pipeline_config"]);
+
+    // boxes walk through
+    $found = false;
+    foreach ($config["boxes"] as &$box) {
+      if ($box["type"] == self::JUDGE_TYPE) {
+        $found = true;
+      } else {
+        continue;
+      }
+
+      $box["portsIn"][self::JUDGE_ARGS_PORT] = [
+        "type" => "string[]",
+        "value" => self::JUDGE_ARGS_VAR
+      ];
+      $box["portsIn"][self::CUSTOM_JUDGE_VAR] = [
+        "type" => "file",
+        "value" => self::CUSTOM_JUDGE_VAR
+      ];
+    }
+
+    if ($found === true) {
+      // add box which will download custom judge if any
+      $config["boxes"][] = [
+        "name" => self::CUSTOM_JUDGE_VAR,
+        "type" => "file-in",
+        "portsIn" => [],
+        "portsOut" => [
+          "input" => [
+            "type" => "file",
+            "value" => self::CUSTOM_JUDGE_VAR
+          ]
+        ]
+      ];
+
+      // add custom judge variable
+      $config["variables"][] = [
+        "name" => self::CUSTOM_JUDGE_VAR,
+        "type" => "file",
+        "value" => ""
+      ];
+
+      // add judge args variable
+      $config["variables"][] = [
+        "name" => self::JUDGE_ARGS_VAR,
+        "type" => "string[]",
+        "value" => self::JUDGE_ARGS_REF
+      ];
+
+      // save
+      $this->connection->executeQuery("UPDATE pipeline_config SET pipeline_config = :config WHERE id = :id",
+        ["config" => Yaml::dump($config), "id" => $pipelineConfig["id"]]);
+    }
+  }
+
+  /**
+   * To configurations using judges add judge arguments variable and custom
+   * judge file variable.
```

Comment:
Please, swap those two sentences. I'm not sure whether you can do this in English, but it doesn't read very well regardless of that.

Reply:
This sentence is really badly build. I'll try to reformulate it.